xbmc/windowing/wayland/WinSystemWayland.cpp

```diff
@@ -592,11 +592,21 @@ bool CWinSystemWayland::SetFullScreen(bool fullScreen, RESOLUTION_INFO& res, boo
       ApplyBufferScale(m_scale);
     }
 
-    // FIXME This assumes that the resolution has already been set. Should
-    // be moved to some post-change callback when resolution setting is refactored.
-    if (!m_inhibitSkinReload)
+    if (m_shellSurfaceState.test(IShellSurface::STATE_RESIZING) && !m_resizeSkinReloadTimer.IsRunning())
     {
-      g_application.ReloadSkin();
+      // Reloading the skin is costly -> during resizing (when the size changes quite
```

Comment:
Drop reloading skin here. This is a general issue in guilib. I had a brief look into the code. guilib seems to evaluate listitem height only on load. GUIWindows get the resize messsage and should do required adjustments there: https://github.com/xbmc/xbmc/blob/master/xbmc/guilib/GUIWindow.cpp#L745